### PR TITLE
PT-2261: Update payment module

### PR DESCRIPTION
**Pull request by aizhuli**
## Description
Refactor payment module to implement generic crud services

### CODE REVIEW

File: src/VirtoCommerce.Payment.Data/Services/PaymentMethodsService.cs [hunk 1]
```diff
@@ -44,95 +38,37 @@ public class PaymentMethodsService : IPaymentMethodsService, IPaymentMethodsRegi
...
-            {
-                var paymentMethodEntities = await repository.GetStorePaymentMethodsByIdsAsync(ids);
-                foreach (var paymentMethodEntity in paymentMethodEntities)
-                {
-                    repository.Remove(paymentMethodEntity);
-                }
-                await repository.UnitOfWork.CommitAsync();
-            }
 
-            await _settingManager.DeepRemoveSettingsAsync(paymentMethods);
```
Comment (akak1977) [inline src/VirtoCommerce.Payment.Data/Services/PaymentMethodsService.cs:135]: No equal code.

File: src/VirtoCommerce.Payment.Core/Services/IPaymentMethodsSearchService.cs [hunk 2]
```diff
@@ -1,10 +1,6 @@
-using System.Threading.Tasks;
-using VirtoCommerce.PaymentModule.Core.Model.Search;
-
 namespace VirtoCommerce.PaymentModule.Core.Services
 {
     public interface IPaymentMethodsSearchService
     {
-        Task<PaymentMethodsSearchResult> SearchPaymentMethodsAsync(PaymentMethodsSearchCriteria criteria);
     }
```
Comment (akak1977) [inline src/VirtoCommerce.Payment.Core/Services/IPaymentMethodsSearchService.cs:5]: Removing a method from the interface introduces a compatibility break.

File: src/VirtoCommerce.Payment.Core/Services/IPaymentMethodsService.cs [hunk 3]
```diff
@@ -1,13 +1,6 @@
...
-
 namespace VirtoCommerce.PaymentModule.Core.Services
 {
     public interface IPaymentMethodsService
     {
-        Task<PaymentMethod[]> GetByIdsAsync(string[] ids, string responseGroup);
-        Task<PaymentMethod> GetByIdAsync(string id, string responseGroup);
-        Task SaveChangesAsync(PaymentMethod[] paymentMethods);
-        Task DeleteAsync(string[] ids);
     }
```
Comment (akak1977) [inline src/VirtoCommerce.Payment.Core/Services/IPaymentMethodsService.cs:5]: Removing a method from the interface introduces a compatibility break.

**Review by akak1977 — CHANGES_REQUESTED** [review]

**Comment by mvktsk** [comment]
Actual PR https://github.com/VirtoCommerce/vc-module-payment/pull/29

**Pull request closed without merging**